src/pages/AddCertificate/AddCertificate.tsx

```diff
@@ -0,0 +1,248 @@
+    validRange: '',
+  });
+
+  const validateForm = (): boolean => {
+    const newErrors: NewFormError = {
+      supplier: '',
+      certificateType: '',
+      validFrom: '',
+      validTo: '',
+      validRange: '',
+    };
+
+    console.log('Validating form data:', certificateData);
+
+    if (!certificateData.supplier) newErrors.supplier = 'Supplier is required.';
+    if (!certificateData.certificateType)
+      newErrors.certificateType = 'Certificate type is required.';
+    if (!certificateData.validFrom)
+      newErrors.validFrom = 'Valid from date is required.';
+    if (!certificateData.validTo)
+      newErrors.validTo = 'Valid to date is required.';
+    if (
+      certificateData.validFrom &&
+      certificateData.validTo &&
+      certificateData.validFrom > certificateData.validTo
+    ) {
+      newErrors.validRange =
+        'Valid from date cannot be after the valid to date.';
+    }
+
+    console.log('Validation errors:', newErrors);
+
+    setErrors(newErrors);
+
+    const isValid = Object.values(newErrors).every((error) => error === '');
+    console.log('Form is valid:', isValid);
+
+    return isValid;
+  };
+
+  const handleInput = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>,
+  ) => {
+    const { name, value } = e.target;
+    setCertificateData((prevData) => ({
+      ...prevData,
+      [name]: value,
+    }));
+  };
+
+  const handleDateChange = (name: string, date: Date | null) => {
+    setCertificateData((prevData) => ({
+      ...prevData,
+      [name]: date,
+    }));
+  };
+
+  const handleSave = () => {
+    console.log('Attempting to save...');
+    if (validateForm()) {
+      console.log('Form validated successfully');
+      setCertificateData(certificateData);
+      certificates.push(certificateData);
+      console.log('Certificate added:', certificateData);
+      console.log('Updated certificates array:', certificates);
+
+      setErrors({
+        supplier: '',
+        certificateType: '',
+        validFrom: '',
+        validTo: '',
+        validRange: '',
+      });
+
+      navigate('/machine-learning/example1');
+    } else {
+      console.log('Form validation failed');
+      console.log('Current errors:', errors);
+    }
+  };
+
+  const handleReset = () => {
+    setCertificateData({
+      supplier: '',
+      certificateType: '',
+      validFrom: null,
+      validTo: null,
+    });
+    setPdfPreview(null);
+  };
+
+  const handlePdfUpload = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0];
+    if (file) {
+      const reader = new FileReader();
+      reader.onload = () => {
+        setPdfPreview(reader.result as string);
+      };
+      reader.readAsDataURL(file);
+    }
+  };
+
+  return (
+    <div className="certificate-form-container">
+      <div className="left-side">
+        <div className="row">
+          <label htmlFor="supplier">Supplier</label>
+          <div className="input-wrapper">
```

Comment:
wrap this into component as well please, SupplierLookup or something, together with errors

Reply:
not yet resolved